### ISSUE TITLE
Support context.getFilename()

### ISSUE THREAD

**Issue by ideadapt** [roadhump/SublimeLinter-eslint#28]
I use context.getFilename() in my custom rule. When my rule is called from SublimeLinter this function returns "<text>". When I call eslint from console the return value is as expected.
Has this something todo with this plugin? Maybe with the template_suffix attribute: http://sublimelinter.readthedocs.org/en/latest/linter_attributes.html#tempfile-suffix

**Comment by roadhump** [comment]
I suppose it's because plugin takes a content of file on input, not file itself. Try your rule with `cat file.js | eslint --stdin`, it should be the same way plugin runs.

**Comment by ideadapt** [comment]
I tried your command. It has the same effect as if my rule is invoked from SublimeLinter plugin. 
I added a tempfile_suffix defintion to you `linter.py` file. Output:
```
SublimeLinter: eslint: mainAction.js ['/usr/local/bin/eslint', '--format=compact'] 
SublimeLinter: eslint output:
/var/folders/nc/pl0c_dnd43bb6xl5ljdfh5zh0000gn/T/SublimeLinter3-ideadapt/mainAction.js: line 7, col 8, Error - Strings must use doublequote. (quotes)
```

So I got some tempfiles now. The sad thing is that now my custom .eslintrc is not respected anymore :/ I think this is because eslint is invoked without the -c argument. I didn't find any way to get the projects .eslintrc path. I saw that eslint for TextMate did something like that in: https://github.com/natesilva/javascript-eslint.tmbundle/blob/master/Support/validate.py

Another thing is that tempfile paths only contain the original filename without any project relative path. What I hoped to get is something like `/var/foldes/temp/T/SublimeLinter3-ideadapt/test/module/mainAction.js`

**Comment by roadhump** [comment]
It's a way SublimeLinter lints files without saving while editing. Try to set it to 'lint on save' in SublimeLinter config, I am not sure, but maybe there will be a path to the original file.

The main reason I use stdin in plugin is that it allows to process .eslintrc correctly, extending it from project folder up to user home folder etc. Plugin used tmp files and explicit .eslintrc before eslint can process stdin and it caused difference between how plugin and eslint work.

Still, I can hardly imagine why rule needs to use file name. Maybe it will be easier to use per-file configuration by `/*eslint ...*/` comments, just a suggestion.

**Comment by ideadapt** [comment]
I now found a way to locate .eslintrc, which at least works for my setups (supports projects and open folders). I post the interesting parts of changed `linter.py` file, just if anyone else is interested in:
```python

from SublimeLinter.lint import NodeLinter
import sublime


class ESLint(NodeLinter):

    """Provides an interface to the eslint executable."""
    eslintrc = None
    folders = sublime.active_window().folders()
    project = sublime.active_window().project_data()
    if project:
        eslintrc = project['folders'][0]['path']
    if len(folders):
        eslintrc = folders[0]
    if eslintrc is None:
        eslintrc = '~'
    eslintrc = eslintrc + '/.eslintrc'

    syntax = ('javascript', 'html', 'javascriptnext', 'javascript (babel)', 'javascript (jsx)')
    npm_name = 'eslint'
    cmd = 'eslint --format=compact --config=' + eslintrc
    tempfile_suffix={
        'javascript': 'js'
    }
```

But still, the problem with the shortened tempfile paths still exists :/

The 'lint on save' did not enable the filename. 
Maybe I am on the wrong track, and rules should work without filenames. But since I want my rule only to apply to test files, and work within the editor, on command line and on CI, I haven't found another way, yet :)

**Comment by roadhump** [comment]
If your test files are in separate folder, you can add sub .eslintrc here

**Comment by glenjamin** [comment]
It might be simpler to ask for eslint to provide a `--filename` option that will override the filename when using `--stdin` ?

**Comment by roadhump** [comment]
@glenjamin You can try, but I really don't see valid case for that, except this specific issue.
